lib/admin/WP_Auth0_Admin_Basic.php

```diff
@@ -32,153 +32,120 @@ public function init() {
+			$this->get_dashboard_link( 'applications' )
+		);
 	}
 
-	public function render_client_secret() {
-		$v = $this->options->get( 'client_secret' );
-	?>
-      <input type="text" autocomplete="off" name="<?php echo $this->options->get_options_name(); ?>[client_secret]" id="wpa0_client_secret"  <?php if ( !empty( $v ) ) {?>placeholder="Not visible"<?php } ?> />
-      <div class="subelement">
-        <span class="description"><?php echo __( 'Application secret, copy from your application\'s settings in the', 'wp-auth0' ); ?> <a href="https://manage.auth0.com/#/applications" target="_blank">Auth0 dashboard</a>.</span>
-      </div>
-  <?php
+	public function render_client_secret_b64_encoded( $args = array() ) {
+		$this->render_switch( $args['label_for'], $args['opt_name'] );
+		$this->render_field_description(
+			__( 'Enable this if your Client Secret is base64 enabled. ', 'wp-auth0' ) .
+			__( 'This information is found below the Client Secret field in the ', 'wp-auth0' ) .
+			$this->get_dashboard_link( 'applications' )
+		);
 	}
 
-	public function render_client_secret_b64_encoded() {
-		$v = absint( $this->options->get( 'client_secret_b64_encoded' ) );
+	public function render_client_signing_algorithm( $args = array() ){
+		$value = $this->options->get( $args['opt_name'],  WP_Auth0_Api_Client::DEFAULT_CLIENT_ALG );
+		$this->render_radio_button( $args['label_for'] . '_hs', $args['opt_name'], 'HS256', '', 'HS256' === $value );
+		$this->render_radio_button( $args['label_for'] . '_rs', $args['opt_name'], 'RS256', '', 'RS256' === $value );
+		$this->render_field_description(
+			__( 'This value can be found the Application settings in the ' ) .
+			$this->get_dashboard_link( 'applications' ) .
+			__( ' under Show Advanced Settings > OAuth > "JsonWebToken Signature Algorithm"', 'wp-auth0' )
+		);
+	}
 
-		$this->render_a0_switch( "wpa_client_secret_b64_encoded", "client_secret_b64_encoded", 1, 1 == $v );
-	?>
-				<div class="subelement">
-					<span class="description"><?php echo __( 'Enable if your client secret is base64 enabled.  If you are not sure, check your clients page in Auth0.  Displayed below the client secret on that page is the text "The Client Secret is not base64 encoded.
-	" when this is not encoded.', 'wp-auth0' ); ?></span>
-				</div>
-			<?php
+	public function render_cache_expiration( $args = array() ) {
+		$this->render_text_field( $args['label_for'], $args['opt_name'], 'number' );
+		printf(
+			' <input type="button" id="auth0_delete_cache_transient" value="%s" class="button button-secondary">',
+			__( 'Delete Cache', 'wp-auth0' )
+		);
+		$this->render_field_description( __( 'JWKS cache expiration in minutes (use 0 for no caching)', 'wp-auth0' ) );
+		if ( $domain = $this->options->get( 'domain' ) ) {
+			$this->render_field_description(
+				sprintf( '<a href="https://%s/.well-known/jwks.json" target="_blank">%s</a>',
+					$domain,
+					__( 'View your JWKS here', 'wp-auth0' )
```

Comment:
Would this section show the JWKs contents or what?

Reply:
Links to it, yes.